app/src/main/java/fr/free/nrw/commons/upload/UploadActivity.java

```diff
@@ -309,19 +309,29 @@ public void setRightCardVisibility(boolean visible) {
     }
 
     @Override
-    public void setBottomCardVisibility(@UploadPage int page) {
+    public void setBottomCardVisibility(@UploadPage int page, int uploadCount) {
         if (page == TITLE_CARD) {
             viewFlipper.setDisplayedChild(0);
         } else if (page == CATEGORIES) {
             viewFlipper.setDisplayedChild(1);
+            TextView categoriesSubtitle = viewFlipper.findViewById(R.id.categories_subtitle);
+            setSubtitleVisibility(categoriesSubtitle, uploadCount);
         } else if (page == LICENSE) {
             viewFlipper.setDisplayedChild(2);
             dismissKeyboard();
+            TextView licenseSubtitle = viewFlipper.findViewById(R.id.license_subtitle);
+            setSubtitleVisibility(licenseSubtitle, uploadCount);
         } else if (page == PLEASE_WAIT) {
             viewFlipper.setDisplayedChild(3);
         }
     }
 
+    private void setSubtitleVisibility(TextView subtitle, int uploadCount) {
+        if (subtitle != null && uploadCount > 1) {
+                subtitle.setVisibility(View.VISIBLE);
+        }
```

Comment:
If we're sticking with this, we should probably have an else that makes them not visible if uploadCount is 1 (still with null check).

This is because although they are originally hidden, we might add 2 pictures which shows the subtitles. Later we might decide to remove one (e.g. prompted by too dark/blurry checks) so uploadCount would be just 1 but subtitles still visible with this method.

Reply:
Yes. Will make the advised changes.